### ISSUE TITLE
[Bug]: Force parameter not respected when using model.deploy() for a Charmhub charm

### ISSUE THREAD

**Issue by dragomirp** [juju/python-libjuju#814]
### Description

While `model.deploy()` does have a `force` parameter, it does not seem it is used when deploying charms and the deploy will fail if the series specified is not set by the charm. This works when called through the Juju CLI:
```
> juju deploy weebl --series jammy --force
Located charm "weebl" in charm-hub, revision 6
Deploying "weebl" from charm-hub charm "weebl", revision 6 in channel stable on jammy
```
When using libjuju:
```
>>> await model.deploy("weebl", series="jammy", force=True)
Traceback (most recent call last):
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "<console>", line 1, in <module>
  File "/home/dragop/workspace/pgbouncer-operator/.tox/client-relation-integration/lib/python3.10/site-packages/juju/model.py", line 1606, in deploy
    res = await self.deploy_types[str(url.schema)].resolve(url, architecture, application_name, channel, series, entity_url)
  File "/home/dragop/workspace/pgbouncer-operator/.tox/client-relation-integration/lib/python3.10/site-packages/juju/model.py", line 568, in resolve
    raise JujuError("Series {} not supported for {}. Only {}".format(series, url, supported_series))
juju.errors.JujuError: Series jammy not supported for ch:weebl. Only ['focal']
```

The `force` parameter seems to be used only when deploying bundles.

### Urgency

Annoying bug in our test suite

### Python-libjuju version

2.9.42.1

### Juju version

2.9.42

### Reproduce / Test

```python
from juju.model import Model

model = Model()
await model.connect_current()
await model.deploy("weebl", series="jammy", force=True)
```

**Comment by juanmanuel-tirado** [comment]
Interestingly the `force` argument is not passed to the api call [here](https://github.com/juju/python-libjuju/blob/2795292ed819e46eaee152a8eb00271542cee818/juju/model.py#L1922). We will fix it. Thanks for reporting!

**Comment by dragomirp** [comment]
This was fixed in 2.9.42.2. Thanks for taking care.
